test/socket_recv_test.cpp

```diff
@@ -0,0 +1,244 @@
+#include "socket.hpp"
+#include <gtest/gtest.h>
+#include <thread>
+#include <string>
+#include <algorithm>
+
+using namespace net;
+using namespace std::chrono_literals;
+
+auto msgLen2 = 15000;
```

Comment:
There shouldn't be any global variables, make them const if possible and move them under appropriate function or atleast some namespace.